flytekit/image_spec/image_spec.py

```diff
@@ -104,7 +104,6 @@ def is_container(self) -> bool:
             return os.environ.get(_F_IMG_ID) == self.image_name()
         return True
 
-    @lru_cache
```

Comment:
Remove lru cache since I don't think it makes sense here. 

1. image.exist() => False
2. build image
3. image.exist() => it should return true instead of reading from the cache

Reply:
do we look things up once per task?  or just once per image spec?  if it's looked up once per task for some reason it's okay to have cache (though i feel we should change that behaviro)

Reply:
Once per image spec. Also, we already have a cache at [build](https://github.com/flyteorg/flytekit/blob/1953b73708a63725ab8f9ff5ecc2c455244758a6/flytekit/image_spec/image_spec.py#L260), so we don't need it here.